overlord/ifacestate/ifacemgr.go

```diff
@@ -78,6 +75,9 @@ func (m *InterfaceManager) Init(s *state.State) error {
 }
 
 func (m *InterfaceManager) doConnect(task *state.Task) error {
+	task.State().Lock()
+	defer task.State().Unlock()
```

Comment:
Okay for now, but we can't lock it all later. This will be non-trivial and have multi-phase operations (commands running, hooks on both ends, etc).